File: solutions/javascript/bob/2/bob.js
class Bob {
  hey(message) {
    if (message.trim() === '') { return 'Fine. Be that way!'; }
    else if (message.match(/[A-Z]/)
          && message.toUpperCase() === message) { return 'Whoa, chill out!'; }
    else if (message[message.length - 1] === '?') { return 'Sure.'; }
    return 'Whatever.';
  }
}

export default Bob;

File: solutions/elixir/bob/2/lib/bob.ex
defmodule Bob do
  @spec hey(String.t()) :: String.t()
  def hey(input) do
    has_letters? = input =~ ~r/\p{L}/u
    is_upcase? = has_letters? and String.upcase(input) == input
    trim = String.trim(input)
    is_question? = String.last(trim) == "?"
    cond do
      trim == "" -> "Fine. Be that way!"
      is_upcase? and is_question? -> "Calm down, I know what I'm doing!"
      is_upcase? -> "Whoa, chill out!"
      is_question? -> "Sure."
      true -> "Whatever."
    end
  end
end
